framework/update/helper/swap.cpp

```
@@ -0,0 +1,234 @@
+
+FILE* g_log = nullptr;
+
+void logLine(const std::string& msg)
+{
+    if (g_log) {
+        std::fprintf(g_log, "%s\n", msg.c_str());
+        std::fflush(g_log);
+    }
+}
+
+Args parseArgs(int argc, char** argv)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)